syntax = "proto3";
import "google/protobuf/empty.proto";

message FileMetadata {
    string name = 1;
    int64 size = 2;
    string timestamp = 3;
}

message FileList {
    repeated FileMetadata metadata = 1;
}

message FileContent {
    string name = 1;
    bytes data = 2;
}

message FileRequest {
    string name = 1;
}

message OperationStatus {
    int32 code = 1;
    string message = 2;
}

service FileService {
    rpc ListFiles(google.protobuf.Empty) returns (FileList) {}
    rpc FindFile(FileRequest) returns (FileList) {}
    rpc PutFile(stream FileContent) returns (OperationStatus) {}
    rpc GetFile(FileRequest) returns (stream FileContent) {}
}
